src/components/input/ButtonBase.tsx

```diff
@@ -1,33 +1,53 @@
 import classNames from 'classnames';
+import type { JSX } from 'preact';
 
+import type { BaseProps } from '../../types';
 import { downcastRef } from '../../util/typing';
 
 /**
  * Common props used for Button components.
- *
- * @typedef ButtonCommonProps
- * @prop {boolean} [expanded] - Is the element associated with this button
- *   expanded? (set `aria-expanded`)
- * @prop {never} [aria-expanded] - Use `expanded` prop instead
- * @prop {boolean} [pressed] - Is this button currently "active?" (set
- *   `aria-pressed` or `aria-selected` depending on button `role`)
- * @prop {never} [aria-pressed] - Use `pressed` prop instead
- * @prop {string} [title] - Button title; used for `aria-label` attribute
- * @prop {never} [aria-label] - Use `title` prop instead
- *
+ */
+export type ButtonCommonProps = {
+  /**
+   * Is the element associated with this button expanded?
+   * (sets `aria-expanded` attribute)
+   */
+  expanded?: boolean;
+
+  /**
+   * Is this button currently "active"?
+   * (sets `aria-pressed` or `aria-selected` depending on button `role`)
+   */
+  pressed?: boolean;
+
+  /**
+   * Button title (sets `aria-label` attribute)
+   */
+  title?: string;
+
+  /** Use `expanded` prop instead */
+  'aria-expanded'?: never;
+  /** Use `pressed` prop instead */
+  'aria-pressed'?: never;
+  /** Use `title` prop instead */
+  'aria-label'?: never;
+};
+
+/**
  * HTML attributes accepted by button components. This eliminates conflicting
  * `icon` and `size` attributes.
- *
- * @typedef {Omit<import('preact').JSX.HTMLAttributes<HTMLButtonElement>, 'icon'|'size'>} HTMLButtonAttributes
- *
  */
+export type HTMLButtonAttributes = Omit<
+  JSX.HTMLAttributes<HTMLButtonElement>,
+  'icon' | 'size'
+>;
```

Comment:
Components in this package frequently intersect in `HTMLAttributes`; I acknowledge that at some point we might want to consolidate these types.

Reply:
Oh, also: Well done to keep exporting this. (deeply) Unfortunately, these types are used by a component or two in the `client`. There's very much a ticket open to consolidate this package's type exports: https://github.com/hypothesis/frontend-shared/issues/858